### ISSUE TITLE
ks delete ERROR strconv.Atoi

### ISSUE THREAD

**Issue by jlewi** [ksonnet/ksonnet#272]
I'm trying to delete the K8s resources associated with one of my ks components. When I run delete I get an error.

```
ks delete gke -c iap
ERROR strconv.Atoi: parsing "7+": invalid syntax
```
Am I doing something wrong?

I don't think its a problem with my config because the following works

```
$ ks show gke -c iap > /tmp/manifests.yaml
$ ks delete /tmp/manifests.yaml 
ERROR Environment '/tmp/manifests.yaml' does not exist
$ kubectl delete -f /tmp/manifests.yaml 
service "iap-jupyter" deleted
ingress "iap-jupyter-load-balancer" deleted
```
Here's the version info

```
$ks version
ksonnet version: dev-2017-12-15T06:22:08-0800
jsonnet version: v0.9.5
client-go version: v1.6.8-beta.0+$Format:%h$
```

**Comment by bryanl** [comment]
I constructed a simple test to see if we had a regression in our component deleter and didn't see anything obvious. WOuld it be possible to show the jsonnet for the the `iap` component?

**Comment by jlewi** [comment]
Thanks Bryan.

It should all be here
https://github.com/jlewi/kubeflow-rl/blob/master/rl-app/components/jupyter-iap.jsonnet

But its probably changed since I reported the bug. 

If you weren't able to reproduce it; I'd suggest closing this for now as can't reproduce and I'll reopen it with more information if I hit it again.

**Comment by phantomnat** [comment]
I have the same problem.
ran `ks apply` and `ks delete` work fine on the Azure cluster.
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.11", GitCommit:"b13f2fd682d56eab7a6a2b5a1cab1a3d2c8bdd55", GitTreeState:"clean", BuildDate:"2017-11-25T18:34:52Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.10", GitCommit:"bebdeb749f1fa3da9e1312c4b08e439c404b3136", GitTreeState:"clean", BuildDate:"2017-11-03T16:31:49Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

but when I ran GKE I got 
```
$ ks delete gke
ERROR strconv.Atoi: parsing "7+": invalid syntax

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.11", GitCommit:"b13f2fd682d56eab7a6a2b5a1cab1a3d2c8bdd55", GitTreeState:"clean", BuildDate:"2017-11-25T18:34:52Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.11-gke.1", GitCommit:"3500f53730c1fea7b57901977df165c3eb317bce", GitTreeState:"clean", BuildDate:"2017-12-08T18:05:07Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```
Maybe the server minor version cloud be the problem?

**Comment by phantomnat** [comment]
My mistake. It fixed in #254 and will come out in version 0.9.0.
